### PR TITLE
LSO: initial commit for the Learning Sequence Object

### PR DESCRIPTION
**Pull request by klees**
Contains the following features (most notably):

* create learning sequence from (some) repo objects, set post conditions
* standard repo functionality: copy, paste, link, ...
* member management with two local roles (admin, learner), including member
  galery, mails, subscribe, unsubscribe
* permissions for admins and members
* learning progress, including settings
* learner view using the new Kiosk Mode Service
* post news to News Service when LSO goes online
* import/export

### CODE REVIEW

File: Services/Contact/classes/class.ilContactGUI.php [hunk 1]
```diff
@@ -128,6 +128,13 @@ public function executeCommand()
 				$this->ctrl->setReturn($this, "showContacts");
 				$this->ctrl->forwardCommand(new ilMailSearchGroupsGUI());
 				break;
+
+			case 'ilmailsearchlearningsequencegui':
```
Comment (mjansenDatabay) [inline Services/Contact/classes/class.ilContactGUI.php]: I do not find a class `ilMailSearchLearningSequenceGUI`. I think these changes here are not necessary.

Comment (klees) [reply Services/Contact/classes/class.ilContactGUI.php]: Thx. @nhaagen, please have a look.

File: Modules/LearningSequence/classes/Xml/class.ilLearningSequenceXMLParser.php [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+			case "lp_type":
+				$this->lp_settings["lp_type"] = trim($this->cdata);
+				break;
+			case "lp_mode":
+				$this->lp_settings["lp_mode"] = trim($this->cdata);
+				break;
+			case "admin_role_ops":
+				$this->role_ops["admin_role_ops"][] = trim($this->cdata);
+				break;
+			case "member_role_ops":
```
Comment (smeyer-ilias) [inline Modules/LearningSequence/classes/Xml/class.ilLearningSequenceXMLParser.php]: Please do not import role permissions. 
A newly created objects should be created using the standard permissions of their context.

**Review by smeyer-ilias — CHANGES_REQUESTED** [review]
In general LGTM.

**Pull request merged**